saleor/graphql/account/mutations/staff/customer_create.py

```diff
@@ -63,7 +65,44 @@ def save(cls, info: ResolveInfo, instance, cleaned_input):
         )
 
         instance.search_document = prepare_user_search_document_value(instance)
-        instance.save(update_fields=["search_document"])
+
+        """
+        Prevent race condition when saving customer.
+
+        This is a hacky solution that catches DB error, to deduct that entity was created before.
+
+        Normally we would use get_or_create here, but it will lose reference - get_or_create
+        returns new instance of model, but original "instance" is being returned in mutation.
+
+        This limitation comes from the Base/Model mutation. We can solve this e.g. by returning
+        actual instance from save()
+
+        https://linear.app/saleor/issue/EXT-2162
+        """
+        try:
+            with transaction.atomic():
+                instance.save()
```

Comment:
You have:

```python
cls.save_default_addresses(
            cleaned_input=cleaned_input, user_instance=instance, save_user=True
        )
```

This is called first, meaning the instance will always be saved to the DB before your logic is executed. It also opens up the potential for a race condition between `save_default_addresses` and a different `create/register-user` mutation.

To avoid this, your newly added logic should be executed first, followed by the actions on the addresses. 

I also recommend wrapping your new logic inside a method, like `_save`, to keep the "hack" contained in one place. This will make it easier to replace with `get_or_create` once `BaseMutation.save` is fixed.